xdsl/dialects/memref_stream.py

```diff
@@ -397,6 +398,9 @@ class GenericOp(IRDLOperation):
     Indices into the `outputs` that correspond to the initial values in `inits`.
     """
 
+    doc: StringAttr | None = opt_prop_def(StringAttr)
+    library_call: StringAttr | None = opt_prop_def(StringAttr)
```

Comment:
I would lean towards this:
```suggestion
    doc = prop_def(StringAttr)
    library_call = prop_def(StringAttr)
```
How is an empty doc and empty library call functionally different from None?

Reply:
Then it's no longer optional and all current tests fail with `Operation does not verify: property doc expected`. I think we want to keep them optional as in mlir? I just copied this from the `linalg.generic` implementation

Reply:
You're right, this is good